packages/contracts/src/manager/libraries/LibAppStorage.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {LibDiamond} from "./LibDiamond.sol";
+import "../access/AccessControl.sol";
+import {AccessControlStorage} from "../libraries/AccessControlStorage.sol";
+
+bytes32 constant GOVERNANCE_TOKEN_MINTER_ROLE =
+    keccak256("GOVERNANCE_TOKEN_MINTER_ROLE");
+bytes32 constant GOVERNANCE_TOKEN_BURNER_ROLE =
+    keccak256("GOVERNANCE_TOKEN_BURNER_ROLE");
+bytes32 constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+bytes32 constant CREDIT_NFT_MANAGER_ROLE = keccak256("CREDIT_NFT_MANAGER_ROLE");
+bytes32 constant STAKING_MANAGER_ROLE = keccak256("STAKING_MANAGER_ROLE");
+bytes32 constant INCENTIVE_MANAGER_ROLE = keccak256("INCENTIVE_MANAGER");
+bytes32 constant GOVERNANCE_TOKEN_MANAGER_ROLE =
+    keccak256("GOVERNANCE_TOKEN_MANAGER_ROLE");
+
+struct AppStorage {
+    address twapOracleAddress;
+    address creditNFTAddress;
+    address dollarTokenAddress;
+    address creditNFTCalculatorAddress;
+    address dollarMintCalculatorAddress;
+    address stakingShareAddress;
+    address stakingContractAddress;
+    address stableSwapMetaPoolAddress;
+    address curve3PoolTokenAddress; // 3CRV
+    address treasuryAddress;
+    address governanceTokenAddress;
+    address sushiSwapPoolAddress; // sushi pool UbiquityDollar-GovernanceToken
+    address masterChefAddress;
+    address formulasAddress;
+    address creditTokenAddress;
+    address creditCalculatorAddress;
+    mapping(address => address) _excessDollarDistributors;
+}
+
+library LibAppStorage {
+    function diamondStorage() internal pure returns (AppStorage storage ds) {
+        assembly {
+            ds.slot := 0
+        }
+    }
+}
+
+contract Modifiers is AccessControl {
+    AppStorage internal s;
+
+    modifier onlyOwner() {
+        LibDiamond.enforceIsContractOwner();
+        _;
+    }
+
```

Comment:
I would add a non reentrant modifier here cos we will need it sooner or later and we can have these fields added to the appstorage struct 
```
   // reentrancy guard
    uint256 NOT_ENTERED;
    uint256 ENTERED;
    uint256 reentrancyStatus;
```

Reply:
> I would add a non reentrant modifier here cos we will need it sooner or later and we can have these fields added to the appstorage struct

Maybe